From 0f19b4f36f6d441c240b5465d9a027d6bfc74f87 Mon Sep 17 00:00:00 2001
From: Ofer Heifetz <oferh@marvell.com>
Date: Tue, 14 Nov 2017 10:04:02 +0200
Subject: [PATCH 0611/1051] crypto: inside-secure: adjust firmware path

New ap810 uses new eip197d which needs different firmware then
the cp110 eip197b, to differentiate between the two firmwares,
a new directory was added to the firmware path.

Change-Id: Id8e7ce1bd1ec2d09c14d977d27943ccc1f5736c0
Signed-off-by: Ofer Heifetz <oferh@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/crypto/inside-secure/safexcel.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/crypto/inside-secure/safexcel.c b/drivers/crypto/inside-secure/safexcel.c
index 09077e0950bd..bc46489f17d1 100644
--- a/drivers/crypto/inside-secure/safexcel.c
+++ b/drivers/crypto/inside-secure/safexcel.c
@@ -141,7 +141,8 @@ static void eip197_write_firmware(struct safexcel_crypto_priv *priv,
 
 static int eip197_load_firmwares(struct safexcel_crypto_priv *priv)
 {
-	const char *fw_name[] = {"ifpp.bin", "ipue.bin"};
+	static const char * const fw_name[] = {"eip197/197b/ifpp.bin",
+					       "eip197/197b/ipue.bin"};
 	const struct firmware *fw[FW_NB];
 	char fw_path[31], *dir = NULL;
 	int i, j, ret = 0, pe;
-- 
2.17.1

